### PR TITLE
(fix) Fix support for latest nightlies

### PR DESCRIPTION
**Pull request by zmbush**
Tested with rustc 1.1.0-nightly (21f278a68 2015-04-23) (built 2015-04-24)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,4 +9,6 @@ license = "MIT"
 
 [dependencies]
 iron = "*"
-
+time = "*"
+typemap = "*"
```
Comment (reem) [inline Cargo.toml]: typemap is re-exported at `iron::typemap`

File: examples/default.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 extern crate iron;
 extern crate logger;
 
-use iron::{Chain, ChainBuilder, Iron, IronResult, Request, Response};
+use iron::{Chain, Iron, IronResult, Request, Response};
```
Comment (reem) [inline examples/default.rs]: Use the prelude `use iron::prelude::*;` :)

**Comment by zmbush** [comment]
@reem, I've updated the PR

File: src/format.rs [hunk 3]
```diff
@@ -33,7 +33,7 @@ impl Default for Format {
     /// green for 200s, yellow for 300s, red for 400s, and bright red for 500s.
     fn default() -> Format {
         fn status_color(_req: &Request, res: &Response) -> Option<color::Color> {
-            match *res.status.as_ref().unwrap_or(&NotFound) as u16 / 100 {
+            match (*res.status.as_ref().unwrap_or(&NotFound)).to_u16() / 100 {
```
Comment (reem) [inline src/format.rs]: We should use the `class` method from hyper: http://hyperium.github.io/hyper/hyper/status/enum.StatusCode.html#method.class

File: src/format.rs [hunk 4]
```diff
@@ -99,7 +99,7 @@ impl Format {
     pub fn new(s: &str, colors: Vec<FormatColor>, attrs: Vec<FormatAttr>)
             -> Option<Format> {
 
-        let mut parser = FormatParser::new(s.chars().peekable(),
+        let parser = FormatParser::new(s.chars().peekable(),
                                            colors.into_iter(),
```
Comment (reem) [inline src/format.rs]: Can you reindent these lines so they match up like before?

File: src/format.rs (hunk 3 above)
Comment (zmbush) [inline src/format.rs]: iron first needs to reexport StatusClass

**Comment by zmbush** [comment]
@reem, I updated the PR, but I had to change the iron dependency to be git rather than a crates.io. Once a new iron is released, then we can switch back to crates.

**Comment by zmbush** [comment]
@reem Updated to support mainline iron. I think this should be ready to go now.

**Comment by zmbush** [comment]
@reem, build is passing now.

**Comment by reem** [comment]
Looks great to me. I updated the travis config to get this building, and am quite excited to land it.

**Pull request merged**